docs/kubernetes/cross-cluster.md

```diff
@@ -0,0 +1,177 @@
+    lost+found
+    testfile
+    ```
+
+    Switch context to `ClusterA`. Check that the deployments in the `ClusterA`, pointing to the same filestore instance, can access the newly created file.
+    ```console
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-kssrl -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-lw2l2 -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-sxrnd -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+    ```
+
+5. Deletion of workloads and the volume resources.
+
+    Since the same filestore instance is shared across clusters, and the workload in one cluster is unaware of the filestore instance being shared by a different cluster, care must be taken by the user to ensure that no running workload is pointing filestore isntance at the time of deletion. Deletion of the filestore instance can be initiated from any cluster.
+
+    Delete the pods in all the cluster
+    ```console
+    $ kubectl delete deployment web-server-deployment-cluster-a
+    deployment.apps "web-server-deployment-cluster-a" deleted
+    ```
+
+    Ensure the PersistentVolumeClaim (PVC) still exists in `ClusterA`
+    ```console
+    $ kubectl get pvc
+    NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+    test-pvc-fs   Bound    pvc-f5a0d4b1-79ff-4ef1-baef-018557eafb00   1Ti        RWX            csi-filestore   39m
+    ```
+
+    Switch to `clusterB` and delete pods.
+    ```console
+    $ kubectl delete deployment web-server-deployment-cluster-b
+    deployment.apps "web-server-deployment-cluster-b" deleted
+    ```
+
+    Ensure PVC still exists
+    ```console
+    $ kubectl get pvc test-pvc-fs
+    NAME          STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+    test-pvc-fs   Bound    pre-pv   1Ti        RWX            csi-filestore   16
+    ```
+
+    At this point no running pod in any cluster is using the filestore instance. We can proceed with deletion of the PVC. In this example we choose to delete the PVC from the `clusterA` first. Since the PV has a Retain policy of `Delete`, this should trigger the deletion of PersistentVolume, and this should delete the filestore instance. For more details into retain policy see [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaiming)
+
+    ```console
+    $ kubectl delete pvc test-pvc-fs
+    $ kubectl get pvc
+    No resources found in default namespace.
+    $ kubectl get pv
+    No resources found in default namespace.
+    ```
+
+    Switch context to `ClusterB` and delete the PVC. In this case, the driver in `ClusterB` will detect will detect the backing filestore instance is already deleted and CSI delete call will succeed, and PVC and PV will be removed.
```

Comment:
Add explicit deletion of PV as appropriate.

Reply:
Done